### PR TITLE
Add google_authentication plugin

### PR DESCRIPTION
**Pull request by krida2000**

### CODE REVIEW

File: src/android/app/build.gradle [hunk 1]
```diff
@@ -70,4 +72,8 @@ flutter {
 
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation platform('com.google.firebase:firebase-bom:26.7.0')
+    implementation 'com.google.firebase:firebase-analytics'
```
Comment (extreamer-lynx) [inline src/android/app/build.gradle:76]: Why you use direbase-analytics, com.google.guava:listenablefuture?

File: src/lib/widgets/screens/user_info_screen.dart [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                          await context.read<AuthenticationBloc>().logout();
+                          Navigator.of(context)
+                              .pushReplacement(_routeToSignInScreen());
+                        } catch (_) {}
+                      },
+                      child: Padding(
+                        padding: EdgeInsets.only(top: 8.0, bottom: 8.0),
+                        child: Text(
+                          'Sign Out',
+                          style: TextStyle(
```
Comment (extreamer-lynx) [inline src/lib/widgets/screens/user_info_screen.dart:146]: move it to MaterialApp widget

File: src/features/google_authentication/lib/bl/authentication_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+import 'package:rxdart/rxdart.dart';
+import 'package:logger/logger.dart';
+
+class AuthenticationBloc {
+  final StreamSink<String>? errorSink;
+
+  AuthenticationBloc({
+    this.errorSink,
```
Comment (extreamer-lynx) [inline src/features/google_authentication/lib/bl/authentication_bloc.dart:12]: use required to Sink

File: src/features/google_authentication/android/google-services.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
+{
```
[the remark this answers is not in the document]
Comment (krida2000) [reply src/features/google_authentication/android/google-services.json:1]: deleted

File: src/android/app/build.gradle (hunk 1 above)
Comment (krida2000) [reply src/android/app/build.gradle:76]: +

File: src/features/google_authentication/lib/bl/authentication_bloc.dart (hunk 3 above)
Comment (krida2000) [reply src/features/google_authentication/lib/bl/authentication_bloc.dart:12]: +

File: src/lib/widgets/screens/user_info_screen.dart (hunk 2 above)
Comment (krida2000) [reply src/lib/widgets/screens/user_info_screen.dart:146]: +

**Pull request merged**